packages/framework/attributor/src/test/mixinAttributor.spec.ts

```diff
@@ -0,0 +1,201 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { strict as assert } from "assert";
+import {
+    AttachState,
+    IContainerContext,
+    ICriticalContainerError,
+} from "@fluidframework/container-definitions";
+import {
+    MockLogger,
+} from "@fluidframework/telemetry-utils";
+import { IDocumentStorageService } from "@fluidframework/driver-definitions";
+import { MockDeltaManager, MockQuorumClients } from "@fluidframework/test-runtime-utils";
+import { ContainerRuntime } from "@fluidframework/container-runtime";
+import { FluidObject } from "@fluidframework/core-interfaces";
+import { ISequencedDocumentMessage, ISnapshotTree, SummaryType } from "@fluidframework/protocol-definitions";
+import { IProvideRuntimeAttribution, mixinAttributor } from "../mixinAttributor";
+import { Attributor as BaseAttributor, OpStreamAttributor } from "../attributor";
+import { makeLZ4Encoder } from "../lz4Encoder";
+import { AttributorSerializer, chain, deltaEncoder } from "../encoders";
+import { makeMockAudience } from "./utils";
+
+type Mutable<T> = {
+    -readonly[P in keyof T]: T[P]
+};
+
+class Attributor extends BaseAttributor {
+    public get type(): string { return "op" };
+}
+
+describe("mixinAttributor", () => {
+    const clientId = "mock client id";
+    let containerRuntime: ContainerRuntime;
+    const getMockContext = ((): Partial<IContainerContext> => {
+        return {
+            audience: makeMockAudience([clientId]),
+            attachState: AttachState.Attached,
+            deltaManager: new MockDeltaManager(),
+            quorum: new MockQuorumClients(),
+            taggedLogger: new MockLogger(),
+            clientDetails: { capabilities: { interactive: true } },
+            // eslint-disable-next-line @typescript-eslint/no-throw-literal
+            closeFn: (error?: ICriticalContainerError): void => { if (error) { throw error; } },
+            updateDirtyContainerState: (_dirty: boolean) => { },
+        };
+    });
+
+    const AttributingContainerRuntime = mixinAttributor([
+        ["op", (runtime, entries) => new OpStreamAttributor(runtime.deltaManager, runtime.getAudience(), entries)]
+    ]);
+
+    it("Attributes ops", async () => {
+        const context = getMockContext() as IContainerContext;
+        containerRuntime = await AttributingContainerRuntime.load(
+            context,
+            [],
+            undefined, // requestHandler
+            {}, // runtimeOptions
+        );
+
+        const maybeProvidesAttribution: FluidObject<IProvideRuntimeAttribution> = containerRuntime as FluidObject;
```

Comment:
any suggestions on alternate pattern to expose the fact that the container runtime implements this functionality? seems like the FluidObject pattern is mostly targeted at components and it's therefore a little odd that it's done so this way right now.

Reply:
this is fixed I believe. Moving the impl to a different object allows placing it on the scope, which is more idiomatic usage. Tests don't need to do weird things--they look like typical FluidObject usage.